lib/Core/SpecialFunctionHandler.cpp

```diff
@@ -936,48 +1010,133 @@ void SpecialFunctionHandler::handleMarkPersistent(ExecutionState &state,
 
   if (name.length() == 0) {
     name = "unnamed";
-    klee_warning("klee_pmem_mark_persistent: renamed empty name to \"unnamed\"");
+    klee_warning("klee_pmem_mark_persistent: name parameter empty, using \"unnamed\"");
+  }
+
+  ref<Expr> addr = arguments[0];
+  ref<Expr> size = arguments[1];
+  uint64_t realSize = 0;
+  if (ConstantExpr *ce = dyn_cast<ConstantExpr>(size.get())) {
+    realSize = ce->getZExtValue();
+  } else {
+    size->dump();
+    klee_error("Not sure how to handle symbolic size argument yet!");
   }
 
   Executor::ExactResolutionList rl;
-  executor.resolveExact(state, arguments[0], rl, "mark_persistent");
+  for (uint64_t offset = 0; offset < realSize; offset += PersistentState::MaxSize) {
+    ref<Expr> offsetExpr = ConstantExpr::create(offset, Expr::Int64);
+    ref<Expr> ptrExpr = AddExpr::create(addr, offsetExpr);
+    executor.resolveExact(state, ptrExpr, rl, "mark_persistent");
+  }
```

Comment:
I'm curious why you're doing it this way. `Executor` already has a `resolve` function that looks at all possibly relevant MemoryObjects.

Reply:
Ah, actually it's in `state.addressSpace`. I'll change this, thanks!